File: python/1910_remove_all_occurrences_of_a_substring.py
class Solution:
    def removeOccurrences(self, s: str, part: str) -> str:
        stack = []
        substring_length = len(part)
        lp = list(part)
        for char in s:
            stack.append(char)
            if stack[-substring_length:] == lp:
                for i in range(substring_length):
                    stack.pop()
        if stack[-substring_length:] == lp:
            for i in range(substring_length):
                stack.pop()
        return "".join(stack)

File: rust/src/1910_remove_all_occurrences_of_a_substring.rs
impl Solution {
    pub fn remove_occurrences(s: String, part: String) -> String {
        let part_length = part.len();
        let mut stack: Vec<char> = Vec::with_capacity(s.len());
        for letter in s.chars() {
            stack.push(letter);
            let top_of_stack: String = stack.iter().rev().take(part_length).rev().collect();
            if top_of_stack == part {
                for _ in (0..part_length) {
                    stack.pop();
                }
            }
        }
        let top_of_stack: String = stack.iter().rev().take(part_length).rev().collect();
        if top_of_stack == part {
            for _ in (0..part_length) {
                stack.pop();
            }
        }
        return stack.into_iter().collect();
    }
}
